CSharpFunctionalExtensions/Maybe/Maybe.cs

```diff
@@ -90,9 +95,9 @@ private Maybe(T value)
 
         public static implicit operator Maybe<T>(Maybe value) => None;
 
-        public static Maybe<T> From(T obj)
+        public static Maybe<T> From(T obj, IEqualityComparer<T> equalityComparer = null)
```

Comment:
It's not backward compatible. This code will not compile anymore. 

```csharp
Foo(Maybe<int>.From)

void Foo(Func<int, Maybe<int>> selector) 
{
} 
```

Let's add a new overload instead.